packages/xstate-wallet/src/browser-wallet.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  BN,
+  calculateChannelId,
+  checkThat,
+  isSimpleEthAllocation,
+  Objective,
+  SignedState,
+  Uint256,
+  Zero
+} from '@statechannels/wallet-core';
+import _, {Dictionary} from 'lodash';
+
+import {ChannelStoreEntry} from './store/channel-store-entry';
+import {logger} from './logger';
+import {Store} from './store';
+import {ChainWatcher, ChannelChainInfo} from './chain';
+
+export type Message = {
+  objectives: Objective[];
+  signedStates: SignedState[];
+};
+type Response = Message & {deposit?: boolean};
+
+type Funding = {amountOnChain: Uint256; status?: 'DEPOSITED'};
+type OnNewMessage = (message: Message) => void;
+
+type DepositInfo = {
+  depositAt: Uint256;
+  myDeposit: Uint256;
+  totalAfterDeposit: Uint256;
+  fundedAt: Uint256;
+};
+
+export class BrowserWallet {
+  private constructor(
+    private onNewMessage: OnNewMessage,
+    private chain = new ChainWatcher(),
+    public store = new Store(chain),
+    private registeredChannels = new Set<string>(),
+    // TODO: probably should be stored in a more permanent storage
+    private channelFunding: Dictionary<Funding> = {}
+  ) {}
+
+  private async init(): Promise<BrowserWallet> {
+    await this.store.initialize();
+    return this;
+  }
+
+  static async create(onNewMessage: OnNewMessage): Promise<BrowserWallet> {
+    return new BrowserWallet(onNewMessage).init();
+  }
+
+  async incomingMessage(payload: Message): Promise<Message> {
+    let response: Message = {
+      objectives: [],
+      signedStates: []
+    };
+    // Store any new objectives
+    const payloadObjective = payload.objectives?.[0];
+    if (!payloadObjective) {
+      logger.info('No incoming objectives');
+    } else {
+      await this.store.addObjective(payloadObjective);
+    }
```

Comment:
Is this meant to be permanent? It only stores _one_ new objective, not _all_ new objectives.

Reply:
This is not meant to be permanent. This implementation is the quick and dirty implementation to get the interop tests to pass and unblock https://github.com/statechannels/statechannels/issues/3401 and https://github.com/statechannels/statechannels/issues/3400.